OpenSim/Tools/MarkerPlacer.cpp

```diff
@@ -242,6 +243,47 @@ bool MarkerPlacer::processModel(Model* aModel,
     /* Load the static pose marker file, and average all the
     * frames in the user-specified time range.
     */
+    TimeSeriesTableVec3 staticPoseTable{aPathToSubject + _markerFileName};
+    const auto& timeCol = staticPoseTable.getIndependentColumn();
+    auto numRowsInRange = std::count_if(timeCol.cbegin(),
+                                        timeCol.cend(),
+                                        [&] (const double& val) {
+                                            return val >= _timeRange[0] &&
+                                                   val <= _timeRange[1];
+                                        });
+    const auto avgRow = staticPoseTable.averageRow(_timeRange[0],
+                                                   _timeRange[1]);
+    for(int r = staticPoseTable.getNumRows() - 1; r >= 0; --r) {
+        if(staticPoseTable.getIndependentColumn()[r] >= _timeRange[0] &&
+           staticPoseTable.getIndependentColumn()[r] <= _timeRange[1]) {
+            if(numRowsInRange > 1) {
+                staticPoseTable.removeRowAtIndex(r);
+                --numRowsInRange;
+            } else {
+                staticPoseTable.updRowAtIndex(r) = avgRow;
+            }
+        }
+    }
+    OPENSIM_THROW_IF(!staticPoseTable.hasTableMetaDataKey("Units"),
+                     Exception,
+                     "MarkerPlacer::processModel -- Marker file does not have "
+                     "'Units'.");
+    Units
+    staticPoseUnits{staticPoseTable.getTableMetaData<std::string>("Units")};
+    double scaleFactor = staticPoseUnits.convertTo(aModel->getLengthUnits());
+    OPENSIM_THROW_IF(SimTK::isNaN(scaleFactor),
+                     Exception,
+                     "Model has unspecified units.");
+    if(std::fabs(scaleFactor - 1) >= SimTK::Eps) {
+        for(unsigned r = 0; r < staticPoseTable.getNumRows(); ++r)
+            staticPoseTable.updRowAtIndex(r) *= scaleFactor;
+
+        staticPoseUnits = aModel->getLengthUnits();
+        staticPoseTable.removeTableMetaDataKey("Units");
+        staticPoseTable.addTableMetaData("Units",
+                                         staticPoseUnits.getAbbreviation());
+    }
+    
     MarkerData* staticPose = new MarkerData(aPathToSubject + _markerFileName);
     staticPose->averageFrames(_maxMarkerMovement, _timeRange[0], _timeRange[1]);
     staticPose->convertToUnits(aModel->getLengthUnits());
```

Comment:
I am not sure I understand this comment.

The function `TimeSeriesTable::averageRow()` is being tested to make sure it returns the *average* row in the specified time-interval. It is pretty simple function if I understand correctly. The added snippet in `MarkerPlacer::processModel` replicates the operations on `MarkerData`. 
The future PR will replace rest of the use of MarkerData in MarkerPlacer and rest of the code.